tools/docker/builder/rdkb/toolchain/bitbake.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+###############################################################
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2020 Tomer Eliyahu (Intel)
+# This code is subject to the terms of the BSD+Patent license.
+# See LICENSE file for more details.
+###############################################################
+
+scriptdir="$(cd "${0%/*}"; pwd)"
+rootdir="${scriptdir%/*/*/*/*/*}"
+
+. ${rootdir}/tools/docker/functions.sh
+
+usage() {
+    echo "usage: $(basename $0) [-hrv]"
+    echo "  options:"
+    echo "      -h|--help - show this help menu"
+    echo "      -r|--rdkb-root - path to rdkb root (default is PWD)"
+    echo "      -i|--build-image - Only build image"
+    echo "      -v|--verbose - verbosity on"
+    echo ""
+}
+
+build_image() {
+    docker build --tag rdkb-builder "$scriptdir/"
+}
+
+main() {
+
+    OPTS=`getopt -o 'hir:v' --long help,image,rdkb-path:,verbose -n 'parse-options' -- "$@"`
+
+    if [ $? != 0 ] ; then err "Failed parsing options." >&2 ; usage; exit 1 ; fi
+
+    eval set -- "$OPTS"
+
+    while true; do
+        case "$1" in
+            -h | --help)               usage; exit 0; shift ;;
+            -i | --image)              build_image; exit 0; shift ;;
+            -r | --rdkb-root)          RDKB_ROOT="$2"; shift; shift ;;
+            -v | --verbose)            VERBOSE=true; shift ;;
+            -- ) shift; break ;;
+            * ) err "unsupported argument $1"; usage; exit 1 ;;
+        esac
+    done
+
+    build_image
+    # Default docker arguments
+    docker_args="\
+     --workdir=$PWD \
+     -e HOME=${HOME} \
+     -e RDKB_ROOT=${RDKB_ROOT}
+     -e USER=${SUDO_USER:-${USER}} \
+     -e USER_ID=${SUDO_UID:-$(id -u)} \
+     -e GROUP_ID=${SUDO_GID:-$(id -g)} \
+     -v ${HOME}:${HOME} \
+     -v ${RDKB_ROOT}:${RDKB_ROOT} \
```

Comment:
Do you really need both $RDKB_ROOT and the whole $HOME to be bind-mounted?
I don't see $HOME used directly, is it used by some of the commands run later?

Reply:
.gitconfig for example. We need the home directory mounted for working with the real user instead of duplicating the home folder.